src/java.base/share/classes/sun/security/x509/BasicConstraintsExtension.java

```diff
@@ -158,10 +163,8 @@ public BasicConstraintsExtension(Boolean critical, Object value)
       */
      public String toString() {
          String pathLenAsString;
-         if (pathLen < 0) {
-             pathLenAsString = " undefined";
-         } else if (pathLen == Integer.MAX_VALUE) {
-             pathLenAsString = " no limit";
+         if (pathLen < 0 || pathLen == Integer.MAX_VALUE) {
+             pathLenAsString = " unconstained";
```

Comment:
Typo: s/unconstained/unconstrained/

But I actually prefer the words "no limit" as that is what [RFC 5280, section 4.2.1.9](https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.9) uses, so please restore those words. You can use that term for the "undefined" case as well.